diff --git a/node_modules/react-native-infinite-pager/src/index.tsx b/node_modules/react-native-infinite-pager/src/index.tsx
index e0bfa4a..12ff1d9 100644
--- a/node_modules/react-native-infinite-pager/src/index.tsx
+++ b/node_modules/react-native-infinite-pager/src/index.tsx
@@ -154,6 +154,7 @@ function InfinitePager(
   const orientation = vertical ? "vertical" : "horizontal";
 
   const pageWidth = useSharedValue(width || 0);
+  const isProgrammaticSet = useSharedValue(false);
   const pageHeight = useSharedValue(height || 0);
   const pageSize = vertical ? pageHeight : pageWidth;
 
@@ -214,15 +215,19 @@ function InfinitePager(
 
       if (index < minIndex || index > maxIndex) return;
 
+      isProgrammaticSet.value = true;
       if (options.animated) {
         const animCfg = {
           ...DEFAULT_ANIMATION_CONFIG,
           ...animCfgVal.value,
         } as WithSpringConfig;
 
-        translate.value = withSpring(updatedTranslate, animCfg);
+        translate.value = withSpring(updatedTranslate, animCfg, () => {
+          isProgrammaticSet.value = false;
+        });
       } else {
         translate.value = updatedTranslate;
+        isProgrammaticSet.value = false;
       }
     }
   );
@@ -263,7 +268,7 @@ function InfinitePager(
       return Math.round(pageAnim.value);
     },
     (cur, prev) => {
-      if (cur !== prev) {
+      if (cur !== prev && !isProgrammaticSet.value) {
         runOnJS(onPageChangeInternal)(cur);
       }
     },
